src/main/java/io/pravega/connectors/hadoop/PravegaInputFormatBuilder.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.pravega.connectors.hadoop;
+
+import com.google.common.base.Preconditions;
+import org.apache.hadoop.conf.Configuration;
+
+/**
+ * A pravega hadoop input connector builder.
+ *
+ */
+public class PravegaInputFormatBuilder extends PravegaBuilder<PravegaInputFormatBuilder> {
+
+    public PravegaInputFormatBuilder() {}
+
+    public PravegaInputFormatBuilder(Configuration conf) {
+        super(conf);
+    }
+
+    @Override
+    public void check() {
+        String[] paramsToCheck = new String[]{
+            PravegaConfig.INPUT_SCOPE_NAME,
+            PravegaConfig.INPUT_STREAM_NAME,
+            PravegaConfig.INPUT_URI_STRING,
+            PravegaConfig.INPUT_DESERIALIZER};
+
+        for (String param : paramsToCheck) {
+            Preconditions.checkArgument(!this.settings.getOrDefault(param, "").isEmpty());
+        }
+    }
+
+    @Override
+    public PravegaInputFormatBuilder withScope(String scopeName) {
+        this.setString(PravegaConfig.INPUT_SCOPE_NAME, scopeName);
+        return builder();
+    }
+
+    @Override
+    public PravegaInputFormatBuilder forStream(String streamName) {
+        this.setString(PravegaConfig.INPUT_STREAM_NAME, streamName);
+        return builder();
+    }
+
+    @Override
+    public PravegaInputFormatBuilder withURI(String uri) {
+        this.setString(PravegaConfig.INPUT_URI_STRING, uri);
+        return builder();
+    }
+
+    @Override
+    public PravegaInputFormatBuilder withDeserializer(String className) {
+        this.setString(PravegaConfig.INPUT_DESERIALIZER, className);
+        return builder();
+    }
+
+    @Override
+    public PravegaInputFormatBuilder startPositions(String startPos) {
```

Comment:
Shouldn't the argument to `startPositions` and `endPositions` be a `StreamCut`?

Reply:
Why is the method called `startPositions` rather than `startPosition`?

Reply:
StreamCut is transparent to end users. they just need to save the positions info in string format. Plural is used to follow the client API endStreamCut.asImpl().getPositions().

Reply:
I don't understand the argument, @fpj what is your take?

Reply:
discussed before

https://github.com/pravega/hadoop-connectors/pull/30/commits/1286c3bc58f383b3a62b26fd104a8b7e5c154a2a#r203123879

Reply:
I raised the exact same issue, it feels more natural to me that it is a stream cut, but @yangb8 argues that it doesn't make sense. I'm still a bit confused about it.

Reply:
I think the expectation is that the application indeed gets the end position out of a stream cut. This code is transforming a stream cut into a JSON object:

https://github.com/pravega/hadoop-connectors/blob/a171316c1ccbd0eead6a64259c2f571231e786f0/src/main/java/io/pravega/connectors/hadoop/PravegaInputFormat.java#L140

Is this a hack to get around the fact that the stream cut API does not export it in a JSON format? If so, then we should create a Pravega issue and work on it.

Reply:
I just refined my code to replace the positions (segment/offset pairs in StreamCut) by the Base64 format of SteamCut. So now it'll make sense to use singular rather than plural. 

https://github.com/pravega/pravega/issues/2490 was opened before to trace the enhancement of textual representation of a StreamCut.